### PR TITLE
bcda-4514 Feature: Enable eob v2

### PR DESCRIPTION
**Pull request by SMLuthi**
### Fixes [BCDA-4514](https://jira.cms.gov/browse/BCDA-4514)

We want to support v2 endpoints for all resources (Patient, Coverage, and EoB) which BFD has created endpoints for. Currently, we do not have EoB enabled for our Patient and Group/All endpoints.

### Proposed Changes

Add `ExplanationOfBenefits` to our our supported resources for v2. Update our smoke tests to include v2 EoB tests (and Patient all resources tests as well as Group v2 tests). Update our postman tests to allow for EoB 202s.

### Change Details

- Add `ExplanationOfBenefits` to v2 resources.
- Updated Smoke tests for v2 calls.
- Update Postman tests for v2 calls.

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

Allow Patient and Group calls to be made in v2 for the EoB resources.

### Feedback Requested

I've changed the current v2 endpoint postman tests to now make successful checks. I've also added tests for `Patient` and `Coverage`. However, there are a lot more v1 calls that could maybe be ported to v2 as well. Could be a future ticket?

### CODE REVIEW

**Comment by msnook** [comment]
What will be needed on the `bcda-ops` side?  I suspect some of the tests that are included in the `Jenkinsfile.smoke_test` will need to be updated.
https://github.com/CMSgov/bcda-ops/blob/master/jenkins_files/Jenkinsfile.smoke_test#L68-L72

Anything else?  Do the new Postman tests correctly handle the `v2Disabled` flag (for `prod`)?
https://github.com/CMSgov/bcda-ops/blob/master/jenkins_files/Jenkinsfile.smoke_test#L531

**Comment by SMLuthi** [comment]
@msnook Thanks for showing me that section in the jenkinsfile smoke tests. You're correct, that will need to be updated along side this to include EOB. The postman tests, however, do correctly handle the `v2Disabled` flag.

File: bcda/api/v2/api_test.go [hunk 1]
```diff
@@ -429,8 +429,8 @@ func (s *APITestSuite) TestResourceTypes() {
 		{"Supported type - Patient", []string{"Patient"}, http.StatusAccepted},
 		{"Supported type - Coverage", []string{"Coverage"}, http.StatusAccepted},
 		{"Supported type - Patient,Coverage", []string{"Patient", "Coverage"}, http.StatusAccepted},
-		{"Unsupported type - EOB", []string{"ExplanationOfBenefit"}, http.StatusBadRequest},
-		{"Unsupported type - default", nil, http.StatusBadRequest},
+		{"Unsupported type - EOB", []string{"ExplanationOfBenefit"}, http.StatusAccepted},
```
Comment (katyasoup) [inline bcda/api/v2/api_test.go:432]: Supported type now? :)

**Review by katyasoup — APPROVED** [review]
LGTM!

**Review by katyasoup — APPROVED** [review]

**Review by CosmicSyntax — APPROVED** [review]
LGTM!

**Pull request merged**